multi-agent/lib/mas/resources/field_encryption.py

```
@@ -0,0 +1,112 @@
+            read_only_hint = hints.get("read_only", {})
+            if read_only_hint.get("read_only") is False:
+                configurable.add(field_name)
+            if "secret" in hints:
+                sensitive.add(field_name)
+        return configurable, sensitive
+
+    def encrypt_fields(
+        self,
+        cfg_dict: dict,
+        model_cls: type,
+        category: str = None,
```

Ruff (0.15.21)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)